tests/helpers/boring_model.py

```diff
@@ -11,12 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Optional
-
 import torch
-from torch.utils.data import DataLoader, Dataset, IterableDataset, Subset
+from torch.utils.data import Dataset, IterableDataset
+
+from pytorch_lightning.demos.boring_classes import BoringDataModule, BoringModel, ManualOptimBoringModel, RandomDataset
```

Comment:
This import could be removed so all tests point directly to these. But that will be a huge PR

Reply:
why are we keeping the one in demos? is it relevant to the package?

Reply:
My best guess is that @williamFalcon wanted a class in the examples that could be directly imported by the users (hence the need to live in the package)